### PR TITLE
Utility/example for deleting empty project versions older than <d> days

### PR DESCRIPTION
**Pull request by AR-Calder**
- Added new example for deleting empty project versions older than a given number of days
- Made datetime conversion utilities more robust
- Swapped own-implementations for builtin functions where appropriate

Was needing to write something like this anyway, though #173 gave me a little extra motivation :) 

Note: I've not been able to fully test this script - please give it a go and let me know if everything looks right.

### CODE REVIEW

File: blackduck/Utils.py [hunk 1]
```diff
@@ -14,38 +14,35 @@
...
+        
+
+def timespan(days_ago, from_date=datetime.now(), delta=timedelta(weeks=1)):
     curr_date = from_date - timedelta(days=days_ago)
     while curr_date < from_date:
-        yield curr_date.isoformat('T', 'seconds')
+        yield curr_date
         curr_date += delta
 
-def min_iso8601():
```
Comment (skiyooka) [inline blackduck/Utils.py:41]: While I am all for deleting code, are these methods that you just recently added e.g. this year? Generally, deleting or changing method signatures are considered as breaking backwards-compatibility in an API.

Comment (AR-Calder) [reply blackduck/Utils.py:41]: None of these were in HubInstance - the only use they had was in demo_client, though I'll update that to use the two new methods.

File: examples/client/delete_empty_project_versions.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    base_url=args.base_url,
+    token=access_token,
+    verify=args.verify
+)
+
+max_age = datetime.now() - timedelta(days=args.days)
+
+for project in bd.get_resource('projects'):
+    # skip projects younger than max age
+    if to_datetime(project.get('createdAt')) > max_age: continue
```
Comment (skiyooka) [inline examples/client/delete_empty_project_versions.py:36]: I'm getting an exception here: TypeError: can't compare offset-naive and offset-aware datetimes

File: blackduck/Utils.py (hunk 1 above)
Comment (skiyooka) [reply blackduck/Utils.py:41]: Okay I'm fine with deleting them.

**Comment by skiyooka** [comment]
You may find https://github.com/blackducksoftware/sage/blob/master/sage_version_activity_to_csv.py interesting as it creates a .csv with one line per project version. The .csv includes all columns I can think of containing decision criteria for deletion. The intended pipeline is to run Sage, then the above script, then [filter_activity.py](https://github.com/blackducksoftware/sage/blob/master/filter_activity.py) to trim the list to the stuff one just wants to delete.

File: examples/client/delete_empty_project_versions.py (hunk 2 above)
Comment (AR-Calder) [reply examples/client/delete_empty_project_versions.py:36]: A possible solution could be to create something like this:

```
def to_local_datetime(date):
    return to_datetime(date).astimezone(tz=tzlocal())   

def get_local_datetime():
    return datetime.now(tz=tzlocal())
```

Normally I'd avoid creating wrappers for stl functions but in this case I don't think it is obvious how to get a timezone aware datetime given  `datetime.now()` and even `datetime.utcnow()` are timezone naive. Thoughts?

Comment (skiyooka) [reply examples/client/delete_empty_project_versions.py:36]: In [sage_version_activity_to_csv.py](https://github.com/blackducksoftware/sage/blob/master/sage_version_activity_to_csv.py) I used: from dateutil.parser import isoparse
Would that work?

Comment (AR-Calder) [reply examples/client/delete_empty_project_versions.py:36]: I'll take a look at that tonight

**Pull request closed without merging**